test/integration/ControllerSolidlyLPStrat-optimism-test.js

```diff
@@ -0,0 +1,328 @@
+    expect(await WETH.balanceOf(controller.address)).to.be.equal(0)
+    // less than 0.1 because swap is not exact
+    expect((await WETH.balanceOf(strat.address)).lt(1e17 + '')).to.be.equal(true)
+    expect(await velo.balanceOf(strat.address)).to.be.equal(0)
+
+    const otherStrat = await deploy(
+      'ControllerSolidlyLPStrat',
+      WETH.address,
+      controller.address,
+      '0x9c12939390052919aF3155f41Bf4160Fd3666A6f', // Velodrome router
+      owner.address,
+      '0x101D5e5651D7f949154258C1C7516da1eC273476', // Gauge
+      '0xFd7FddFc0A729eCF45fB6B12fA3B71A575E1966F'  // LP
+    )
+
+    await setupStrat(otherStrat)
+
+    await expect(controller.setStrategy(otherStrat.address)).to.emit(
+      controller, 'NewStrategy'
+    ).withArgs(strat.address, otherStrat.address)
+
+    expect(await controller.balanceOf(bob.address)).to.be.equal('' + 100e18)
+    expect(await WETH.balanceOf(controller.address)).to.be.equal(0)
+    // less than 0.1 because swap is not exact
+    expect((await WETH.balanceOf(strat.address)).lt(1e17 + '')).to.be.equal(true)
+    expect(await velo.balanceOf(strat.address)).to.be.equal(0)
+
+    await waitFor(strat.unpause())
+
+    await expect(controller.setStrategy(strat.address)).to.emit(
+      controller, 'NewStrategy'
+    ).withArgs(otherStrat.address, strat.address)
+
+    expect(await controller.balanceOf(bob.address)).to.be.equal('' + 100e18)
+    expect(await WETH.balanceOf(controller.address)).to.be.equal(0)
+    // less than 0.1 because swap is not exact
+    expect((await WETH.balanceOf(strat.address)).lt(1e17 + '')).to.be.equal(true)
+    expect(await velo.balanceOf(strat.address)).to.be.equal(0)
+  })
+
+  itIf(hre.network.config.network_id === 10, 'Rebalance', async function () {
+    const newBalance = ethers.utils.parseUnits('100')
+
+    await setWethBalanceFor(bob.address, '100', 3)
+    await WETH.connect(bob).transfer(strat.address, ethers.utils.parseUnits('1'))
+
+    expect(await WETH.balanceOf(strat.address)).to.be.equal(1e18 + '')
+    expect(await velo.balanceOf(strat.address)).to.be.equal(0)
+
+    await strat.rebalance()
+
+    expect((await WETH.balanceOf(strat.address)).lt(1e16 + '')).to.be.equal(true)
+
+    const timestamp = (await ethers.provider.getBlock()).timestamp
+
+    // Since sETH is a proxy contract instead of custom set balance we go with some good old swap =)
+    await WETH.connect(bob).approve(exchange.address, 2e18 + '')
+
+    await waitFor(
+      exchange.connect(bob).swapExactTokensForTokens(
+        2e18 + '',
+        1e18 + '',
+        [{ from: WETH.address, to: seth.address, stable: true }],
+        bob.address,
+        timestamp + 60
+      )
+    )
+
+    await seth.connect(bob).transfer(strat.address, ethers.utils.parseUnits('1'))
+
+    expect(await seth.balanceOf(strat.address)).to.be.gte(1e18 + '')
+
+    await strat.rebalance()
+
+    expect((await seth.balanceOf(strat.address)).lt(1e16 + '')).to.be.equal(true)
```

Comment:
let's add here 
```javascript
oldBalance = await strat.balance()
await rebalance()
expect(await strat.balance).to.be.within(oldBalance + 0.9e18, oldBalance + 1e18)
```
To be sure that the rebalance works as expected. (Same in line 301 in the other rebalance)